src/bgw/job.c

```diff
@@ -184,9 +184,11 @@ get_job_type_from_name(Name job_type_name)
 }
 
 static BgwJob *
-bgw_job_from_tuple(HeapTuple tuple, size_t alloc_size, MemoryContext mctx)
+bgw_job_from_tupleinfo(TupleInfo *ti, size_t alloc_size, MemoryContext mctx)
 {
 	BgwJob *job;
+	bool should_free;
+	HeapTuple tuple = ExecFetchSlotHeapTuple(ti->slot, false, &should_free);
```

Comment:
How does the memory behave before/after this change? I see code where we copy the tuple contents into alloc'ed memory. Could you elaborate on how/why ExecFetchSlotHeapTuple helps here.